UTRAnnotator.pm

```diff
@@ -230,23 +230,39 @@ sub run {
   );
 
 
-  my @output_five_prime_flag=();
-  my @output_five_prime_annotation=();
+  my $output_five_prime_flag;
+  my $output_five_prime_annotation;
 
   foreach my $flag (keys %five_prime_flag){
     if($five_prime_flag{$flag}){
-      push @output_five_prime_flag,$five_prime_flag{$flag};
-      push @output_five_prime_annotation,$five_prime_annotation{$flag};
+      $output_five_prime_flag = $five_prime_flag{$flag};
+      $output_five_prime_annotation = $five_prime_annotation{$flag};
     }
   };
 
-  @output_five_prime_flag=("-") if(!@output_five_prime_flag);
-  @output_five_prime_annotation=("-") if(!@output_five_prime_annotation);
+  my %utr_effect;
 
-  my %utr_effect = (
-        "5UTR_consequence" => (join "&", @output_five_prime_flag),
-        "5UTR_annotation" => (join "&", @output_five_prime_annotation),
-  );
+  if ($self->{config}->{output_format} eq "json" || $self->{config}->{rest}){
+
+    %utr_effect = (
+        "5UTR_consequence" => $output_five_prime_flag,
+        "5UTR_annotation" => $output_five_prime_annotation,
+    );
+
+  } else {
+
+    my $consequence;
+
+    foreach my $value (keys %{$output_five_prime_annotation}){
```

Comment:
Is `output_five_prime_annotation` a list? then you should not have call `keys` here.

Reply:
Updated!